### DISCUSSION TITLE
Error reinforcing mine - Solved by allowing TUS spending

### DISCUSSION THREAD

**Discussion by cameronqj** [coccoinomane/crabada.py#89] [Q&A] [answered]
execution reverted: ERC20: transfer amount exceeds allowance

USER_1_TEAM_1_REINFORCE_STRATEGY="HighestMp" # look in ReinforceStrategyFactory.py for available strategies
USER_1_TEAM_1_TASK="mine"
USER_1_TEAM_1_REINFORCEMENT_TO_PICK="1" # increase to avoid reverted txs at a potential expense of stats value
USER_1_REINFORCEMENT_MAX_PRICE="18" # in TUS - will not reinforce if borrow price is larger than this
USER_1_REINFORCEMENT_MAX_GAS="100" # in gwei - will not reinforce if base fee is larger than this (set to "inf" or delete line for no limit)

Consistently errors out with:

src.common.logger - INFO - Borrowing crab 5315 for mine 5424734 at 6.2 TUS... [BP=192, MP=72]
2022-04-26 16:21:05,632 - src.common.logger - INFO - Borrowing crab 5315 for mine 5424734 at 6.2 TUS... [BP=192, MP=72]
src.common.logger - WARNING - Error reinforcing mine 5424734: execution reverted: ERC20: transfer amount exceeds allowance

I've tried adjussting max gas to 500 / inf and that doesn't seem to make any difference..Any advice?

**Comment by cameronqj** [comment]
Also tried raising USER_1_REINFORCEMENT_MAX_PRICE -- no change.

**Comment by cameronqj** [comment]
Commenting out USER_1_REINFORCEMENT_MAX_GAS still ending up with same result.

**Comment by cameronqj** [comment, marked as answer]
I hadn't authorized TUS to be spent in this wallet. PEBKAC

**Reply by loayei** [reply]
Yeah just do one transaction using the UI to approve TUS then it should work fine

**Comment by coccoinomane** [comment]
Hi @cameronqj !

As @loayei suggested, you have to allow the Crabada's contract to spend TUS for you. The simplest way is to reinforce at least once from the browser on play.crabada.com :-)

Cheers,
Cocco
